client/js/serverstate.js

```diff
@@ -125,9 +125,8 @@ function receiveDeltaFromServer(delta) {
 function receiveStateFromServer(args) {
   mouseTarget = null;
   deltaID = args._meta.deltaID;
-  for(const widget of $a('#room .widget'))
-    if(widget.id != 'enlarged')
-      widgets.get(widget.id).applyRemove();
+  for(const el of $a('[id^=w_]'))
+    widgets.get(unescapeID(el.id.slice(2))).applyRemove();
```

Comment:
Hmm. Can anyone think of a reason why this doesn't just iterate through `widgets`?

Reply:
It used to just remove the HTML element and was changed to `applyRemove` when [I introduced piles](https://github.com/ArnoldSmith86/virtualtabletop/commit/b83b4bafeffefedc6d1c3ad0931deb6078a3aa94#diff-e79126ca55855f229c5a0580c7990110f30632522801e8f1b611a99b2d22d31fL118). So there probably isn't a good reason.

Reply:
So we could just change this to
```js
  for(const widget of widgets)
    widget.applyRemove();
```
Seems like that would be more efficient and easier to read. I wasn't sure why it wasn't that way in the first place, so I was reluctant to change it. If it was just removing the HTML elements originally, then iterating over them directly was probably more efficient at the time.

Reply:
Well, that didn't work (room repeatedly crashes). Looks like the composed widgets from the add widgets screen are actually still in `widgets` but do not have an `applyRemove()` method.

Reply:
In `addCompositeWidgetToAddWidgetOverlay()` (editmode.js) the composite widgets are created then added to widgets with `widgets.set(wi.id, w)`. This is probably needed to correctly apply the widgets to the DOM. I added a loop over widgetsToAdd which removed the widgets from widgets and the empty test room stopped crashing on load; however the crash here when TestCafe set the state to empty between tests returned. I'm not going to dig into this any more in this PR.

Reply:
It's working fine. :man_shrugging: